Frontend/src/app/clusters/[id]/access-management/page.tsx

```diff
@@ -0,0 +1,222 @@
+            </Tag>) : (
+            <Tag color={"volcano"}>
+              {"Inactive"}
+            </Tag>
+          )}
+        </>
+      ),
+    }
+  ];
+
+  const adminsAccessColumns: TableProps<DataType>['columns'] = [...columns,
+    {
+      title: 'Action',
+      key: 'action',
+      render: (_, record) => (
+        <Space size="middle">
+          <Button type="primary" danger size="small" onClick={() => handleUserRemove(record.userID)}>Remove</Button> 
+        </Space>
+      ),
+    }
+  ];
+
+  const getTableData = () => {
+    return clusterUsers.map((data) => {
+      return {
+        key: data._id,
+        userID: data._id,
+        Name: data.fullName,
+        Email: data.email,
+        Role: data.role,
+        Status: data.enabled,
+      };
+    });
+  }
+
+  const handleUserAdd = async (userID: string) => {
+    const user = userData.find((user) => user._id == userID)
+    setClusterUsers([...clusterUsers, user!])
+    try{
+      const res = await addUserToCluster(userID, clusterDetails!._id);
+      if(res.data){
+        console.log("User added successfully")
+      }
+    }catch(err){
+      console.log("failed to user add")
+    }
+  }
+
+  const handleUserRemove = async (userID: string) => {
+    setClusterUsers(clusterUsers.filter((user) => user._id !== userID))
+    try{
+      const res = await removeUserFromCluster(userID, clusterDetails!._id);
+      if(res.data){
+        console.log("User removed successfully")
+      }
+    }catch(err){
+      console.log("failed to user add")
+    }
+  }
+
+  const renderItem = (title: string, userID: string) => ({
+    value: title,
+    label: (
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'space-between',
+        }}
+      >
+        {title}
+        <span>
+           <Button size="small" type="primary" onClick={()=> handleUserAdd(userID)}>Add</Button>
+        </span>
+      </div>
+    ),
+  });
+
+  const [options, setOptions] = React.useState<AutoCompleteProps['options']>([]);
+
+  const handleSearch = (value: string) => {
+    setOptions(() => {
+      const filteredUsers = userData.filter((user) => 
+                        user.fullName.toLowerCase().includes(value.toLowerCase()) || 
+                        user.email.toLowerCase().includes(value.toLowerCase()));
+      return filteredUsers.map((user) => (renderItem(user.email, user._id)));
+    });
+  };
+
+  const fetchClusterDetails = async () => {
+    try{
+      const res = await getCluster(params.id);
+      if(res.data){
+        setClusterDetails(res.data)
+      }
+      dispatch(set_Selected_Cluster(res.data))
+    }catch(err){
+      console.log("Failed to fetch cluster")
+    }
+  }
+
+  const fetchUserData = async () => {
+    try{
+      const res = await getAllUsers();
+      if(res.data){
+        setUserData(res.data)
+        dispatch(set_Users_Cache(res.data))
+      }
+    }catch(err){
+      console.log("Failed to fetch cluster")
```

Comment:
Need to have context to this error message.

Reply:
if needed cluster name and description can be added